cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,423 @@
+
+	if verbose {
+		fmt.Printf("\nLets take a look at the status of your Targeted Nodes...\n\n")
+	}
+
+	for i, node := range *targetsInfo {
+		info := "\t🥸  Node Name: " + node.Name + "\n" +
+			"\t📲 Node Addresses: \n"
+
+		for _, address := range node.Status.Addresses {
+			info += "\t\t⭕️ Type: " + string(address.Type) + "\n" +
+				"\t\tℹ️  Address: " + address.Address + "\n"
+		}
+
+		info += "\t📜 Node Conditions:\n"
+
+		// adding 1 to these states so we can run percentages. Since we have the number
+		// of instances with that specific state (the += 1.0) and we know the total
+		// number of instances, we can easily get a percentage.
+		for _, status := range node.Status.Conditions {
+			info = info +
+				"\t\t⭕️ Type: " + string(status.Type) + "\n" +
+				"\t\tℹ️  Status: " + string(status.Status) + "\n" +
+				"\t\t🤨 Reason: " + status.Reason + "\n\n"
+			percentCollectCondition[string(status.Type)+"/"+string(status.Status)] += 1.0 / float64(len(*targetsInfo))
+		}
+
+		if i < maxtargetshow && verbose {
+			fmt.Print(info)
+		}
+	}
+
+	if verbose {
+		PrintSeparator()
+	}
+
+	percentInfo := "Lets look at the overall status...\n"
+
+	for key, value := range percentCollectCondition {
+		roundedValue := value * 100.00
+		percentInfo += "\tCondition Type/Status: " + key + "\n" +
+			"\tPercent:               " + fmt.Sprint(math.Round(roundedValue*100)/100) + "%\n\n"
+	}
+
+	fmt.Println(percentInfo)
+	PrintSeparator()
+}
+
+var contextCmd = &cobra.Command{
+	Use:   "context",
+	Short: "contextualizes disruption config",
+	Long:  `makes use of kubectl to give a better idea of how big the scope of the disruption will be.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		path, _ := cmd.Flags().GetString("path")
+		kubeconfig, _ = cmd.Flags().GetString("kubeconfig")
+		verbose, _ = cmd.Flags().GetBool("verbose")
+		maxtargetshow, _ = cmd.Flags().GetInt("maxtargetshow")
+		contextualize(path)
+	},
+}
+
+func contextualize(path string) {
+	disruption := ReadUnmarshalValidate(path)
+
+	if len(kubeconfig) == 0 {
+		kubeconfig = filepath.Join(homedir.HomeDir(), ".kube", "config")
+	}
+
+	pods, nodes, err := contextTargetsSize(disruption)
+
+	if err != nil {
+		log.Fatalf("Could not grab context regarding size and names of targets: %v", err)
+	}
+
+	// validate should catch if the disruption level is invalid, safe to assume default else is Node
+	if disruption.Spec.Level == types.DisruptionLevelPod {
+		printPodStatus(pods)
+		printContainerStatus(pods)
+	} else {
+		printNodeStatus(nodes)
+	}
+}
+
+func init() {
+	contextCmd.Flags().String("path", "", "The path to the disruption file to be contextualized.")
+	contextCmd.Flags().String("kubeconfig", "", "The path to your kube configuration directory (.../.kube/config). defaults to ~/.kube/config.")
+	contextCmd.Flags().Bool("verbose", false, "If set, will describe a small set of 10 of your targets. Otherwise it only describes percentages of the group of targets in total.")
```

Comment:
The help text feels slightly confusing? I am surprised to hear that "verbose" leads to describing a small set of targets, though I do see how that works. It does seem that `10` here is wrong, and it's really the value of maxtargetshow?

Reply:
Yeah my thought process behind verbose was that if people are running disruptions on tons of targets (which is usually the case), they don't really want to see individual targets information but what the cluster looks like as a whole. I'll change the comment to specify that 5 is the default which assumes you can change it. Let me know if there is any other confusion in the help text.